napalm_eos/eos.py

```diff
@@ -1090,35 +1093,43 @@ def get_route_to(self, destination='', protocol=''):
         return routes
 
     def get_snmp_information(self):
+        """get_snmp_information() for EOS.  Re-written to not use TextFSM"""
 
-        snmp_information = dict()
-
-        commands = list()
-        commands.append('show running-config | section snmp-server')
-        raw_snmp_config = self.device.run_commands(commands, encoding='text')[0].get('output', '')
-
-        snmp_config = napalm_base.helpers.textfsm_extractor(self, 'snmp_config', raw_snmp_config)
-
-        if not snmp_config:
-            return snmp_information
-
-        snmp_information = {
-            'contact': py23_compat.text_type(snmp_config[0].get('contact', '')),
-            'location': py23_compat.text_type(snmp_config[0].get('location', '')),
-            'chassis_id': py23_compat.text_type(snmp_config[0].get('chassis_id', '')),
+        # Default values
+        snmp_dict = {
+            'chassis_id': '',
+            'location': '',
+            'contact': '',
             'community': {}
         }
 
-        for snmp_entry in snmp_config:
-            community_name = py23_compat.text_type(snmp_entry.get('community', ''))
-            if not community_name:
-                continue
-            snmp_information['community'][community_name] = {
-                'acl': py23_compat.text_type(snmp_entry.get('acl', '')),
-                'mode': py23_compat.text_type(snmp_entry.get('mode', 'ro').lower())
-            }
+        commands = [
+            'show snmp chassis',
+            'show snmp location',
+            'show snmp contact'
+        ]
+        snmp_config = self.device.run_commands(commands, encoding='json')
+        for line in snmp_config:
+            for k, v in line.items():
+                if k == 'chassisId':
+                    snmp_dict['chassis_id'] = v
+                else:
+                    snmp_dict[k] = v
```

Comment:
> However, I don't like that we're doing now four requests to the device in order to solve a small problem.

run_commands is a single request in pyeapi, no matter how many commands are passed. So it's technically only two calls instead of one.  I'm trying to stick with JSON wherever possible to cut down on the amount of screen scraping that has to happen -- "show snmp community" is unfortunately still an unconverted command.  I've informed our account rep that we're interested in seeing the command converted sooner rather than later, which would put us back down to one request instead of two.

In terms of actual performance, I do not have any experience with in-depth Python profiling, but on an example run using "time", the execution difference is very small, and usually outweighed by network delays:

```
[bewing@host:~/napalm-eos] [napalm-eos] develop* ± time python test.py
{u'chassis_id': u'',
 u'community': {u'with-acl': {u'acl': u'ipv4-std-acl', u'mode': u'ro'}},
 u'contact': u'Brandon Ewing',
 u'location': u''}

real    0m0.610s
user    0m0.115s
sys     0m0.014s
[bewing@chcss015:~/napalm-eos] [napalm-eos] develop* ± git checkout iss113
Switched to branch 'iss113'
Your branch is up-to-date with 'origin/iss113'.
[bewing@host:~/napalm-eos] [napalm-eos] iss113* ± time python test.py
{u'chassis_id': u'',
 u'community': {u'blah': {u'acl': u'', u'mode': u'ro'},
                u'public': {u'acl': u'', u'mode': u'ro'},
                u'with-acl': {u'acl': u'ipv4-std-acl', u'mode': u'ro'}},
 u'contact': u'Brandon Ewing',
 u'location': u''}

real    0m0.575s
user    0m0.114s
sys     0m0.016s
```